### PR TITLE
[Config] Remove argument `log_report_frequency`

### PR DESCRIPTION
**Pull request by isratnisa**
Resolves Issue #118 

Removed argument `log_report_frequency`

### CODE REVIEW

**Comment by isratnisa** [comment]
@thvasilo I have removed the `_do_report` param from `sagemaker_tracker.py` file due to its dependency to `log_report_frequency`. Can you please check if that's alright?

File: python/graphstorm/tracker/sagemaker_tracker.py [hunk 1]
```diff
@@ -52,7 +42,7 @@ def log_metric(self, metric_name, metric_value, step, force_report=False):
         force_report: bool
             If true, report the metric
         """
-        if force_report or self._do_report(step):
```
Comment (classicsong) [inline python/graphstorm/tracker/sagemaker_tracker.py:55]: do_report is useful when we are sending log to a system like mlflow or tensorboard.

Comment (classicsong) [reply python/graphstorm/tracker/sagemaker_tracker.py:55]: SageMaker Tracker also use this to decide whether to print the train step.

Comment (isratnisa) [reply python/graphstorm/tracker/sagemaker_tracker.py:55]: I see. We have to keep the argument for the `do_report` argument from SageMaker. For GSF, we can just make it an internal argument by disabling `group.add_argument("--log-report-frequency",...)`

Comment (zhjwy9343) [reply python/graphstorm/tracker/sagemaker_tracker.py:55]: What is our final decision for the "--log-report-frequency"?

Comment (classicsong) [reply python/graphstorm/tracker/sagemaker_tracker.py:55]: I would say, let's keep it for now.

Comment (isratnisa) [reply python/graphstorm/tracker/sagemaker_tracker.py:55]: OK. Closing this PR.

**Pull request closed without merging**